pkg/oci/pack.go

```
@@ -0,0 +1,306 @@
+)
+
+// Pack reads a skill directory, validates the SkillCard, creates an OCI image,
+// and stores it in the local OCI layout. It returns the manifest descriptor.
+func (c *Client) Pack(ctx context.Context, skillDir string, opts PackOptions) (ocispec.Descriptor, error) {
+	// 1. Read and parse skill.yaml.
+	skillPath := filepath.Join(skillDir, "skill.yaml")
+	f, err := os.Open(skillPath)
+	if err != nil {
+		return ocispec.Descriptor{}, fmt.Errorf("opening skill.yaml: %w", err)
+	}
+	defer f.Close()
```

golangci-lint (2.11.4)

[error] Error return value of `f.Close` is not checked

(errcheck)